debug_scripts/estimate_epoch_start_time.py

```diff
@@ -119,6 +119,54 @@ def find_epoch_for_timestamp(future_epochs, voting_timestamp):
     return len(future_epochs)
 
 
+def find_best_voting_hour(voting_date_str, future_epochs, target_timezone):
+    WORKING_HOURS_START = 8  # 8:00 UTC
+    WORKING_HOURS_END = 22  # 22:00 UTC
+
+    valid_hours = []
+
+    for hour in range(24):
+        # Construct datetime for each hour of the voting date
+        voting_datetime = datetime.strptime(
+            f"{voting_date_str} {hour:02d}:00:00", '%Y-%m-%d %H:%M:%S')
+        voting_datetime = target_timezone.localize(voting_datetime)
+        voting_timestamp = voting_datetime.timestamp()
+
+        # Find the epoch T in which the voting date falls
+        epoch_T = find_epoch_for_timestamp(future_epochs, voting_timestamp)
+        if epoch_T <= 0:
+            continue  # Voting date is before the first predicted epoch
+
+        # Calculate when the protocol upgrade will happen (start of epoch T+2)
+        protocol_upgrade_epoch_number = epoch_T + 2
+        if protocol_upgrade_epoch_number > len(future_epochs):
+            print(
+                "Not enough future epochs predicted to determine all the protocol upgrade times."
+            )
+            break
+
+        protocol_upgrade_timestamp = future_epochs[protocol_upgrade_epoch_number
+                                                   - 1]
+        protocol_upgrade_datetime = datetime.fromtimestamp(
+            protocol_upgrade_timestamp)
+        upgrade_datetime_utc = protocol_upgrade_datetime.astimezone(pytz.utc)
```

Comment:
Why do we need to convert this datetime to UTC? In `predict_future_epochs`, we kept all `future_epochs` in UTC.

Reply:
you are right, I forgot it from the previous version where future_epochs was converted to the target timezone